azext_edge/tests/edge/support/create_bundle_int/helpers.py

```diff
@@ -195,62 +205,96 @@ def _check_non_pod_files(workload_types: List[str], required: bool = False):
         _check_non_pod_files(optional_workload_types, required=False)
 
 
+def check_log_for_evicted_pods(file_pods: List[Dict[str, str]], bundle_dir: str):
+    # open the file using bundle_dir and check for evicted pods
+    name_extension_pair = list(set([(file["name"], file["extension"]) for file in file_pods]))
+    with ZipFile(bundle_dir, 'r') as zip:
+        file_names = zip.namelist()
+        for name, extension in name_extension_pair:
+            if extension == "log":
+                # find file path in file_names that has name and extension
+                file_path = next((file for file in file_names if file.endswith(name + ".yaml")), None)
+                if not file_path:
+                    continue
+                with zip.open(file_path) as pod_content:
```

Comment:
if you are reading every file - may as well do it earlier (I can also just adjust the code in a future pr)

Reply:
you said you are going to have a pr contain the file contents soon right?

Reply:
just put in a todo fix here

Reply:
kkk